tests/test_individual.py

```diff
@@ -108,32 +114,33 @@ def test_choose_actions_on_lockdown(self, dir_params, lockdown):
         """
 
         current_lockdown = Hypothesis.read_hypotheses(
-            dir_params,
-            set([lockdown]),
-            "lockdown"
+            dir_params, set([lockdown]), "lockdown"
         )
         # Create an individual with id 0
         individual = Individual.populate(1, dir_params)
 
-        actions, action_probs = individual[0].\
-            choose_actions_on_lockdown(current_lockdown[lockdown])
+        actions, action_probs = individual[0].choose_actions_on_lockdown(
+            current_lockdown[lockdown], rng=np.random.default_rng(seed)
+        )
 
-        assert isinstance(actions, np.ndarray), \
-            'actions should be a numpy array'
-        assert isinstance(action_probs, np.ndarray), \
-            'action_probs should be a numpy array'
+        assert isinstance(actions, np.ndarray), "actions should be a numpy array"
+        assert isinstance(
+            action_probs, np.ndarray
+        ), "action_probs should be a numpy array"
 
-        assert len(actions) == 10, \
-            'actions array should have length 10'
-        assert np.all((actions == 0) | (actions == 1)), \
-            'all actions should be False or True'
+        assert len(actions) == 10, "actions array should have length 10"
+        assert np.all(
+            (actions == 0) | (actions == 1)
+        ), "all actions should be False or True"
 
-        assert np.all((0 <= action_probs) & (action_probs <= 1)), \
-            'all action_probs should be between 0 and 1'
-        assert len(action_probs) == len(actions), \
-            'action_probs and actions should have the same length'
+        assert np.all(
+            (0 <= action_probs) & (action_probs <= 1)
+        ), "all action_probs should be between 0 and 1"
+        assert len(action_probs) == len(
+            actions
+        ), "action_probs and actions should have the same length"
```

Comment:
I see that the logic has not been changed here, but anyway as a comment: I would try to replace the multiple specific assertions with one specific comparison between the output array and the expected. That would check for things like object type, length, and content.

Reply:
I understand the reasoning, but I wonder if your method allows for an informative message on exactly the type of mismatch that occurred and also how to fix it? e.g., knowing that 'action_probs' should be between 0 and 1 is something that I don't see how a generic check between the expected output and the actual output can accomplish. It could only tell you that the numbers don't match, not that your error is that you are not treating the variable as a probability....does this make sense?

Reply:
I think I missed the specific values indeed. But assuming that the result is deterministic, I wonder if you could specify the expected outcome and compare the actual outcome to that, using Numpy's [`assert_array_equal()` function](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html).
My impression is that this would be safer because it checks the outcome, not only the range.

In case the outcome is not strictly deterministic, there is also [`assert_allclose()`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html).